File: qc-core/src/main/java/com/eager/questioncloud/core/domain/user/infrastructure/UserAccountInformationEntity.java
package com.eager.questioncloud.core.domain.user.infrastructure;

import com.eager.questioncloud.core.domain.user.enums.AccountType;
import com.eager.questioncloud.core.domain.user.model.UserAccountInformation;
import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Embeddable
@AllArgsConstructor
@NoArgsConstructor
public class UserAccountInformationEntity {
    private String password;

    private String socialUid;

    @Enumerated(EnumType.STRING)
    private AccountType accountType;

    public static UserAccountInformationEntity from(UserAccountInformation userAccountInformation) {
        return new UserAccountInformationEntity(
            userAccountInformation.getPassword(),
            userAccountInformation.getSocialUid(),
            userAccountInformation.getAccountType());
    }

    public UserAccountInformation toModel() {
        return new UserAccountInformation(password, socialUid, accountType);
    }
}

File: qc-core/qc-user/qc-user-rdb/src/main/kotlin/com/eager/questioncloud/user/entity/UserAccountInformationEntity.kt
package com.eager.questioncloud.user.entity

import com.eager.questioncloud.user.domain.UserAccountInformation
import com.eager.questioncloud.user.enums.AccountType
import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Embeddable
class UserAccountInformationEntity(
    private var password: String? = null,
    private var socialUid: String? = null,
    @Enumerated(EnumType.STRING)
    private var accountType: AccountType
) {
    fun toModel(): UserAccountInformation {
        return UserAccountInformation(password, socialUid, accountType)
    }
    
    companion object {
        fun from(userAccountInformation: UserAccountInformation): UserAccountInformationEntity {
            return UserAccountInformationEntity(
                userAccountInformation.password,
                userAccountInformation.socialUid,
                userAccountInformation.accountType
            )
        }
    }
}
